[RHACS] Improve docs for RHACS OpenShift OAuth provider (#45324)
openshift/openshift-docs: modules/configure-ocp-oauth-identity-provider.adoc

.Prerequisites
* You must have the `AuthProvider` permission to configure identity providers in {product-title-short}.

> * You must have already configured users and groups in {ocp} OAuth server through an identity provider. For details, see link:https://docs.openshift.com/container-platform/4.9/authentication/understanding-identity-provider.html[Understanding identity provider configuration] for additional information.

Reviewer: +1 maybe something like:

s/For details, see Understanding identity provider configuration for additional information./For information on the identity provider requirements, see Understanding identity provider configuration.

> * You must have already configured users and groups in {ocp} OAuth server through an identity provider. For details, see link:https://docs.openshift.com/container-platform/4.9/authentication/understanding-identity-provider.html[Understanding identity provider configuration] for additional information.

Author: Removed suffix, used the suggested formulation.

.Prerequisites
* You must have the `AuthProvider` permission to configure identity providers in {product-title-short}.

> * You must have already configured users and groups in {ocp} OAuth server via an identity provider. Refer to link:https://docs.openshift.com/container-platform/4.9/authentication/understanding-identity-provider.html[Understanding identity provider configuration] for additional information.

Reviewer 2: ```suggestion
* You must have already configured users and groups in {ocp} OAuth server through an identity provider. For details, see link:https://docs.openshift.com/container-platform/4.9/authentication/understanding-identity-provider.html[Understanding identity provider configuration].
```

> 

Reviewer 3: Is this blank line on purpose?

Author: Nope, removed it.

====
For security, Red Hat recommends setting the *Minimum access role* to *None* while you complete setup. Later, you can return to the *Access Control* page to set up more tailored access rules based on user metadata from your identity provider.
====

. To add access rules for users and groups accessing {product-title-short}, use the *Rules* section. For example:
.. To give the *Admin* role to a user called `administrator`, you can use the following key-value pairs to create access rules:
+
[cols="1,1"]
|===

> |Key |Value

Reviewer 4: Do you want to highlight **Key** and **Value** to distinguish them, from the other rows and columns?

====
For security, Red Hat recommends setting the *Minimum access role* to *None* while you complete setup. Later, you can return to the *Access Control* page to set up more tailored access rules based on user metadata from your identity provider.
====
. To add access rules for users and groups accessing {product-title-short}, use the *Rules* section. For example, to give the *Admin* role to a user called `administrator`, select *Name* as *Key*, *administrator* as *Value*, and *Admin* as the *Role*. Use *Add new rule* to add more rules.
. Click *Save*.

.Example values
Given an link:https://docs.openshift.com/container-platform/4.9/authentication/understanding-identity-provider.html[HTPasswd Identity Provider] with a user with name "UserA" and group "GroupA",
the following fields will be available to create access rules:

* "Name"="UserA"
* "Group"="GroupA"

> * "UserID"="<UUID>"

Reviewer 2: . To add access rules for users and groups accessing {product-title-short}, use the *Rules* section. For example:
.. To give the *Admin* role to a user called `administrator`, you can use the following key-value pairs to create access rules:
+
[cols="1,1"]
|===
|Key |Value

|Name
|administrator

|Role
|Admin
|===
.. If you are using the link:https://docs.openshift.com/container-platform/4.9/authentication/understanding-identity-provider.html[HTPasswd Identity Provider] with the username `UserA` that is part of the group `GroupA`, you can use the following key-value pairs to create access rules:
+
[cols="1,1"]
|===
|Key |Value

|Name
|UserA

|Group
|GroupA

|UserID
|<UUID>
|===
. Click *Save*.

====
For security, Red Hat recommends setting the *Minimum access role* to *None* while you complete setup. Later, you can return to the *Access Control* page to set up more tailored access rules based on user metadata from your identity provider.
====

> . To add access rules for users and groups accessing {product-title-short}, use the *Rules* section. For example, to give the *Admin* role to a user called `administrator`, select *Name* as *Key*, *administrator* as *Value*, and *Admin* as the *Role*. Use *Add new rule* to add more rules.

Reviewer 3: Please mention that `email` attribute is not set for this auth provider because of https://docs.openshift.com/container-platform/4.9/rest_api/user_and_group_apis/user-user-openshift-io-v1.html

Author: Added!
